Lib/test/test_functools.py

```diff
@@ -2305,6 +2305,26 @@ def _(arg: typing.Iterable[str]):
         ))
         self.assertTrue(str(exc.exception).endswith(msg_suffix))
 
+    def test_invalid_positional_argument(self):
+        @functools.singledispatch
+        def f(x=3):
+            pass
+
+        @functools.singledispatch
+        def g(*args):
+            pass
+
+        @functools.singledispatch
+        def e(**kwds):
+            pass
+
+        msg = 'singledispatch requires at least 1 positional argument'
+        with self.assertRaisesRegexp(TypeError, msg):
+            f()
+        with self.assertRaisesRegexp(TypeError, msg):
+            g()
+        with self.assertRaisesRegexp(TypeError, msg):
+            e()
```

Comment:
Test seems too redundant for now.
TypeError happens always when no positional arguments.
How the generic function defined is not important.